### PR TITLE
[Merged by Bors] - feat(CategoryTheory): infer connectedness of comma categories from finality/initiality of the functor

### PR DESCRIPTION
**Pull request by javra**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/IsConnected.lean [hunk 1]
```diff
@@ -115,11 +115,15 @@ universe v₂ u₂
...
-theorem isConnected_iff_of_final (F : C ⥤ D) [CategoryTheory.Functor.Final F] :
-    IsConnected C ↔ IsConnected D := by
+theorem isConnected_iff_of_final (F : C ⥤ D) [F.Final] : IsConnected C ↔ IsConnected D := by
   rw [isConnected_iff_colimit_constPUnitFunctor_iso_pUnit.{max v u v₂ u₂} C,
     isConnected_iff_colimit_constPUnitFunctor_iso_pUnit.{max v u v₂ u₂} D]
   exact Equiv.nonempty_congr <| Iso.isoCongrLeft <|
     CategoryTheory.Functor.Final.colimitIso F <| constPUnitFunctor.{max u v u₂ v₂} D
 
+/-- The domain of an initial functor is connected if and only if its codomain is connected. -/
+theorem isConnected_iff_of_initial (F : C ⥤ D) [F.Initial] : IsConnected C ↔ IsConnected D := by
```
[the remark this answers is not in the document]
Comment (javra) [reply Mathlib/CategoryTheory/Limits/IsConnected.lean]: We could also put those into a new file, wdyt?

File: Mathlib/CategoryTheory/Comma/Final.lean [hunk 2]
```diff
@@ -72,6 +74,16 @@ instance initial_snd [L.Initial] : (snd L R).Initial := by
   haveI : (snd L R).op.Final := final_of_natIso (opFunctorCompFst _ _)
   apply initial_of_final_op
 
+/-- `Comma L R` with `L : A ⥤ T` and `R : B ⥤ T` is connected if `R` is final and `A` is
+connected.-/
```
Comment (TwoFX) [inline Mathlib/CategoryTheory/Comma/Final.lean:78]:
```suggestion
connected. -/
```

File: Mathlib/CategoryTheory/Comma/Final.lean [hunk 3]
```diff
@@ -72,6 +74,16 @@ instance initial_snd [L.Initial] : (snd L R).Initial := by
...
+instance isConnected_comma_of_final [IsConnected A] [R.Final] : IsConnected (Comma L R) := by
+  rwa [isConnected_iff_of_final (fst L R)]
+
+/-- `Comma L R` with `L : A ⥤ T` and `R : B ⥤ T` is connected if `L` is initial and `B` is
+connected.-/
```
Comment (TwoFX) [inline Mathlib/CategoryTheory/Comma/Final.lean:83]:
```suggestion
connected. -/
```

File: Mathlib/CategoryTheory/Comma/Final.lean [hunk 4]
```diff
@@ -4,13 +4,15 @@ Released under Apache 2.0 license as described in the file LICENSE.
...
-import Mathlib.CategoryTheory.Limits.Final
+import Mathlib.CategoryTheory.Limits.IsConnected
 import Mathlib.CategoryTheory.Grothendieck
 
 /-!
 # Finality of Projections in Comma Categories
 
 We show that `fst L R` is final if `R` is and that `snd L R` is initial if `L` is.
+As a corollary, we show that `Comma L R` with `L : A ⥤ T` and `R : B ⥤ T` is connected if `R` is
+final and `A` is connected.
```
Comment (TwoFX) [inline Mathlib/CategoryTheory/Comma/Final.lean:15]: Could you add a `References` section to this module doc pointing at Prop. 3.4.3 of Kashiwara-Schapira?

File: Mathlib/CategoryTheory/Limits/IsConnected.lean (hunk 1 above)
[the remark this answers is not in the document]
Comment (TwoFX) [reply Mathlib/CategoryTheory/Limits/IsConnected.lean]: You can do so if you like, but I don't think it's strictly necessary.

File: Mathlib/CategoryTheory/IsConnected.lean [hunk 5]
```diff
@@ -259,6 +259,10 @@ theorem isPreconnected_of_isPreconnected_op [IsPreconnected Jᵒᵖ] : IsPreconn
 theorem isConnected_of_isConnected_op [IsConnected Jᵒᵖ] : IsConnected J :=
   isConnected_of_equivalent (opOpEquivalence J)
 
+variable (J) in
+theorem isConnected_iff_isConnected_op : IsConnected J ↔ IsConnected Jᵒᵖ :=
```
Comment (joelriou) [inline Mathlib/CategoryTheory/IsConnected.lean:263]: I would prefer `IsConnected Jᵒᵖ ↔ IsConnected J`, and we may consider making it a simp lemma.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**